shared/api/container_tty.go

```diff
@@ -0,0 +1,8 @@
+package api
+
+// ContainerTTYControl represents a message on the container tty "control" socket
+type ContainerTTYControl struct {
```

Comment:
I think we're better off making this a separate ContainerConsoleControl as only Args makes sense.
You can't use Command or Signal with the container console.

Reply:
So I'd say:

 - Rename file to container_console.go
 - Rename struct to ContainerConsoleControl
 - Strip struct of everything but Args

Reply:
The reason I didn't call it `container_console.go` is that in a followup branch I'm adding `container_console.go` based on this one and adding the specific bits for cosole connection.

Why would `Command` and `Args` not be used? they're currently used to send  the "window-resize" event, which AFAIU would be appliable to the console too.

Reply:
I could re-add `ContainerExecControl` extending this one with the additional fields.
I thought that removing it wouldn't really break the API as it seems to be used just internally (although it's public because of go package scope). Is there a case where an API user needs to use that struct explicitely?

Reply:
Oh, right, Command and Args make sense, Signal is the one that doesn't.

Reply:
The packages that are considered to be stable API are:
 - client/
 - lxc/config/
 - shared/api/

If someone was to implement something like lxc/exec.go, they'd need to use api.ContainerExecControl, I in fact have two such projects where your change would break my codebase.